src/components/TablePagination.tsx

```diff
@@ -0,0 +1,193 @@
+import React, {
+  ChangeEvent,
+  Children,
+  FC,
+  HTMLAttributes,
+  PropsWithChildren,
+  ReactElement,
+  ReactNode,
+  RefObject,
+  cloneElement,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import classnames from "classnames";
+
+const figureSmallScreen = (descriptionRef: RefObject<HTMLDivElement>) => {
+  const descriptionElement = descriptionRef.current;
+  if (!descriptionElement) {
+    return true;
+  }
+  return descriptionElement.getBoundingClientRect().width < 230;
+};
+
+const renderChildren = (
+  children: ReactNode,
+  dataPropOverride: string,
+  data: unknown[],
+) => {
+  return Children.map(children, (child) => {
+    return cloneElement(child as ReactElement, {
+      [dataPropOverride]: data,
+    });
+  });
+};
+
+const DEFAULT_PAGE_LIMITS = [50, 100, 200];
+const generatePagingOptions = (pageLimits: number[]) => {
+  return pageLimits.map((limit) => ({ value: limit, label: `${limit}/page` }));
+};
+
+interface Props {
+  data: unknown[];
+  itemName?: string;
+  containerClass?: string;
+  displayTitle?: ReactNode;
+  renderBelow?: boolean;
+  dataPropOverride?: string;
+  pageLimits?: number[];
+  dataFiltered?: boolean;
+}
+
+const TablePagination: FC<
+  PropsWithChildren<Props> & HTMLAttributes<HTMLDivElement>
+> = ({
+  data,
+  containerClass,
+  itemName = "item",
+  displayTitle,
+  renderBelow,
+  dataPropOverride = "rows",
+  pageLimits = DEFAULT_PAGE_LIMITS,
+  dataFiltered = false,
+  children,
+  ...divProps
+}) => {
+  const descriptionRef = useRef<HTMLDivElement>(null);
+  const [isSmallScreen, setSmallScreen] = useState(false);
+  const [pageSize, setPageSize] = useState(() => {
+    return generatePagingOptions(pageLimits)[0].value;
+  });
+  const { paginate, currentPage, pageData, totalItems } = usePagination(data, {
+    itemsPerPage: pageSize,
+    autoResetPage: true,
+  });
+
+  useEffect(() => {
+    const handleResize = () => {
+      setSmallScreen(figureSmallScreen(descriptionRef));
+    };
+    window.addEventListener("resize", handleResize);
+    return () => {
+      window.removeEventListener("resize", handleResize);
+    };
+  }, [isSmallScreen]);
+
+  const handlePageSizeChange = (e: ChangeEvent<HTMLSelectElement>) => {
+    paginate(1);
+    setPageSize(parseInt(e.target.value));
+  };
+
+  const getDisplayTitle = () => {
+    if (displayTitle) {
+      return displayTitle;
+    }
+
+    const visibleCount = pageData.length;
+    const filteredText = dataFiltered ? "filtered" : "";
+
+    if (isSmallScreen) {
+      return `${visibleCount} out of ${totalItems}`;
+    }
+
+    if (visibleCount === totalItems && visibleCount > 1) {
+      return `Showing all ${totalItems} ${filteredText} ${itemName}s`;
+    }
+
+    return `Showing ${visibleCount} out of ${totalItems} ${filteredText} ${itemName}${
+      totalItems !== 1 ? "s" : ""
+    }`;
+  };
+
+  const getPageLimitOptions = () => {
+    return generatePagingOptions(pageLimits || DEFAULT_PAGE_LIMITS);
```

Comment:
Why apply defaults here again, when already done above in the component where the props are received?

this seems better to be called directly instead of creating a callback, i.e.

```
const pageLimitOptions = generatePagingOptions(pageLimits);
```

Reply:
good spot, will remove